Detectors/TPC/calibration/src/CalculatedEdx.cxx

```diff
@@ -487,3 +634,65 @@ void CalculatedEdx::loadCalibsFromCCDB(long runNumberOrTimeStamp)
   const o2::base::MatLayerCylSet* matLut = o2::base::MatLayerCylSet::rectifyPtrFromFile(cm.get<o2::base::MatLayerCylSet>("GLO/Param/MatLUT"));
   propagator->setMatLUT(matLut);
 }
+
+void CalculatedEdx::loadCalibsFromLocalCCDBFolder(const char* localCCDBFolder)
+{
+  // set the track topology correction
+  o2::tpc::CalibdEdxTrackTopologyPol calibTrackTopology;
+  calibTrackTopology.loadFromFile(TString::Format("%s/TPC/Calib/TopologyGainPiecewise/snapshot.root", localCCDBFolder), "ccdb_object");
+  mCalibCont.setPolTopologyCorrection(calibTrackTopology);
+
+  // set the gain map
+  TFile* gainMapFile = TFile::Open(TString::Format("%s/TPC/Calib/PadGainFull/snapshot.root", localCCDBFolder));
+  if (!gainMapFile->IsZombie()) {
+    std::cout << "Using file: " << gainMapFile->GetName() << std::endl;
+    o2::tpc::CalDet<float>* gainMap = (o2::tpc::CalDet<float>*)gainMapFile->Get("ccdb_object");
+    const float minGain = 0;
+    const float maxGain = 2;
+    mCalibCont.setGainMap(*gainMap, minGain, maxGain);
+  }
+
+  // set residual gain map
+  TFile* gainMapResidualFile = TFile::Open(TString::Format("%s/TPC/Calib/PadGainResidual/snapshot.root", localCCDBFolder));
+  if (!gainMapResidualFile->IsZombie()) {
+    std::cout << "Using file: " << gainMapResidualFile->GetName() << std::endl;
+    o2::tpc::CalDet<float>* gainMapResidual = (o2::tpc::CalDet<float>*)gainMapResidualFile->Get("ccdb_object");
```

Comment:
The object in `PadGainResidual` is a `std::unordered_map<string,o2::tpc::CalDet<float> >`
You will need to access the member `GainMap` inside.